gateway/src/shard/processor/impl.rs

```diff
@@ -407,12 +407,24 @@ impl ShardProcessor {
 
             if let Err(source) = self.process().await {
                 #[cfg(feature = "tracing")]
-                tracing::debug!(
-                    shard_id = self.config.shard()[0],
-                    shard_total = self.config.shard()[1],
-                    "processing incoming event failed: {:?}",
-                    source,
-                );
+                match &source.kind {
+                    ProcessErrorType::EventTypeUnknown { .. } => {
+                        tracing::debug!(
+                            shard_id = self.config.shard()[0],
+                            shard_total = self.config.shard()[1],
+                            "processing incoming event failed: {:?}",
+                            source,
+                        );
+                    }
+                    _ => {
+                        tracing::warn!(
+                            shard_id = self.config.shard()[0],
+                            shard_total = self.config.shard()[1],
+                            "processing incoming event failed: {:?}",
+                            source,
+                        );
+                    }
+                }
```

Comment:
```suggestion
                if source.kind == ProcessErrorType::EventTypeUnknown {
                    tracing::debug!(
                        shard_id = self.config.shard()[0],
                        shard_total = self.config.shard()[1],
                        "processing incoming event failed: {:?}",
                        source,
                    );
                } else {
                    tracing::warn!(
                        shard_id = self.config.shard()[0],
                        shard_total = self.config.shard()[1],
                        "processing incoming event failed: {:?}",
                        source,
                    );
                }

Reply:
Alt use `matches!()` since ProcessErrorType has no `#[derive(PartialEq)]`

Reply:
i thought we where trying to avoid using macro's in twilight?

Reply:
`matches` is okay, since there's not really something precisely similar without a macro